src/modeler/loadFiles/readLibraries.cpp

```diff
@@ -61,7 +63,7 @@ static ModelerStudy::SystemModel::Library loadSingleLibrary(const fs::path& file
 
     try
     {
-        return ModelConverter::convert(libraryObj);
+        return ModelConverter::convert(libraryObj, optimConfig);
```

Comment:
```c++
return ModelConverter::convert(libraryObj, optimConfig);
```

Here we're supposed to **convert** yaml object into model business objects.
But we do something else :
- we check that elements in **optimConfig** already exist in libraries
- we merge **optimConfig** (more precisely its **location** part)  and libraries to build model variables (business objects)

It makes conversion more complicated than it should.
We could make the **check** and the **merge** from the 2 yaml sources in a previous step, and therefore simplify the **convert** step.

So we would have : 
```c++

try
{
    mergeOptimConfigToLibs(optimConfig, libraryObj);
}
catch(...)
...

try
{
    return ModelConverter::convert(libraryObj);
}
catch (...)
...
```

Reply:
Following a discussion with @guilpier-code , I agree that merging all required characteristics from the `optimConfig` and `libraryObj` seems clearer as operations and responsibilities are more clearly delimited. Especially, the `optimConfig` will be enriched in the future to include new characteristics and concepts, and possibly merging methods will be added and / or updated. With the design of Guillaume, it will be easy to identify which part of the code to modify and how, whereas the current design forces us to go at lower level and possibly make modifications in several methods which may not be clearly identified.